### ISSUE TITLE
Can't serialize maps with non-String keys on wasm

### ISSUE THREAD

**Issue by dgulotta** [johanhelsing/bevy_pkv#62]
Attempting to serialize a HashMap with non-string keys results an error in wasm (but works on native).  This appears to be an intentional limitation of serde_json.  It would be nice if bevy_pkv used a storage method that did not have this limitation.

**Comment by johanhelsing** [comment]
Yes, that sounds like an unnecessary limitation. It would break data for any users upgrading, but I think I'm ok with that.

**Comment by johanhelsing** [comment]
There is also a request to store using serde_json on native for (human readableness #55 ), perhaps it would make sense to have it as a feature flag, affecting both platforms.

**Comment by dgulotta** [comment]
It would be nice to have a lower-level interface that deals with u8 slices and strs directly, in case users want to choose their serialization method.  The higher-level interface that uses rmp-serde or serde-json could exist on top of that.

**Comment by analytik** [comment]
Has this been resolved, or is there some workaround for this? I'm getting the same issue, with `Json(Error("key must be a string", line: 0, column: 0))`

bevy 0.18, bevy_pkv 0.15.0

**Comment by dgulotta** [comment]
I have been working around the issue by using msgpack + base64 to seralize data to a string before passing it to bevy-pkv.

**Comment by analytik** [comment]
Thanks! In the meantime I found a different approach, using the crate `serde_with` in case someone will find this issue:

```rust
use serde_with::serde_as;
use std::collections::HashMap;

// the order here is important!
#[serde_as]
#[derive(serde::Serialize, serde::Deserialize, Default)]
pub struct ActivityStats {
    pub id: usize,
    #[serde_as(as = "Vec<(_, _)>")]
    pub year: HashMap<YearStats, u64>,
    // ...
}

#[derive(serde::Serialize, serde::Deserialize, Default, Eq, PartialEq, Hash)]
pub struct YearStats {
    pub year: i16,
}
```
